styleguide/global/global.js

```diff
@@ -32,6 +32,28 @@ jQuery(document).ready(($) => {
 })(jQuery); // to protect and scope the JQuery alias = $
 
 
+// This plugin can be used to anchor scroll to anything. Usage:
+// 1) Give your link the "rsAnchor-scroll" class
+// 2) Give the href of your link a hashed id (ie href="#someID")
+// 3) Set that id to the element you want scrolled to (ie id="someID")
+(($) => {
+  $.fn.anchorScroll = function anchorScroll() {
+    $(this).click(function scrollTo(e) {
+      e.preventDefault();
+      // we need to subtract the top nav or else scrolled section will look cut off.
+      const fixedNav = $('.navbar-fixed-top').outerHeight();
+      const $this = $(this.hash);
+      if (typeof $this.offset() !== typeof undefined && $this.offset() !== false) {
+        $('html, body').animate({
+          scrollTop: ($this.offset().top - fixedNav),
+        }, 500);
```

Comment:
I know this calculation is simple, but if we don't write this to be unit testable now, we set ourselves up for trouble down the road.  

I'd recommend creating a separate function called something like `calculateAnchorScrollTop()`, that takes, top and fixed nav as parameters.  

Then we need to write unit tests to tests what happens when various values are sent into this function. 

You can use filter-bar as a point of reference for writing the tests...
https://github.com/rackerlabs/zoolander/blob/master/styleguide/derek/global-components/filter-bar/filter-bar.spec.js